test/DomUpdates-test.js

```diff
@@ -1,16 +1,62 @@
 const chai = require('chai');
 import { expect } from 'chai';
 import DomUpdates from '../src/DomUpdates';
+import Sleep from '../src/Sleep';
+import Activity from '../src/Activity';
+import Hydration from '../src/Hydration';
+import UserRepo from '../src/User-repo';
+import User from '../src/User';
 const spies = require('chai-spies');
-// const domUpdates = require('../src/DomUpdates')
 chai.use(spies);
 
-describe('DomUpdates', () => {
-  let domUpdates, currentUser, data, userRepo;
+describe.only('DomUpdates', () => {
+  let domUpdates, currentUser, sleepRepo, hydrationRepo, activityRepo, mostRecentDate, userRepo;
+
+  before(() => {
+    hydrationRepo = new Hydration([
+      {
+        userID: 1,
+        date: '2019/06/15',
+        numOunces: 37,
+      },
+    ]);
+    currentUser = new User({
+      id: 1,
+      name: 'Luisa Hane',
+      address: '15195 Nakia Tunnel, Erdmanport VA 19901-1697',
+      email: 'Diana.Hayes1@hotmail.com',
+      strideLength: 4.3,
+      dailyStepGoal: 10000,
+      friends: [1],
+    });
+    userRepo = new UserRepo([currentUser]);
+    sleepRepo = new Sleep([
+      {
+        userID: 1,
+        date: '2019/06/15',
+        hoursSlept: 6.1,
+        sleepQuality: 2.2,
+      },
+    ]);
+    activityRepo = new Activity([
+      {
+        userID: 1,
+        date: '2019/06/15',
+        numSteps: 3577,
+        minutesActive: 140,
+        flightsOfStairs: 16,
+      },
+    ]);
+
+    mostRecentDate = '2019/06/15';
+  });
 
   beforeEach(() => {
     domUpdates = new DomUpdates();
-    chai.spy.on(domUpdates, ['populateUserWidget'], () => {});
+    global.document = {};
+    chai.spy.on(document, ['getElementById'], () => {
+      return { innerText: '' };
+    });
   });
 
```

Comment:
These are actually pretty thorough tests! Good job!

Reply:
yeah you did a great job on this!